.github/workflows/build_docs.yml

```diff
@@ -0,0 +1,86 @@
+name: Build PR Docs
+
+on:
+  workflow_call:
+    inputs:
+      main-repo:
+        required: true
+        type: string
+        description: 'The main code repository (e.g. napari/napari or napari/docs)'
+      main-ref:
+        required: true
+        type: string
+        description: 'The ref (branch/sha) to use for the main repo (e.g. PR branch or main)'
+      docs-repo:
+        required: true
+        type: string
+        description: 'The docs repository (e.g. napari/docs)'
+      docs-ref:
+        required: true
+        type: string
+        description: 'The ref (branch/sha) to use for the docs repo (e.g. PR branch or main)'
+      make_target:
+        description: "Enter make target: html html-noplot docs slimfast slimgallery"
+        type: string
+        default: "slimfast"
+  workflow_dispatch:
+
+jobs:
+  build-and-upload:
+    name: Build & Upload Docs Artifact
+    runs-on: ubuntu-latest
+    env:
+      XDG_CACHE_HOME: ${{ github.workspace }}/.cache
+    steps:
+      - name: Checkout docs repo
+        uses: actions/checkout@v5
+        with:
+          repository: ${{ inputs.docs-repo }}
+          ref: ${{ inputs.docs-ref }}
+          path: docs
+      - name: Checkout main repo
+        uses: actions/checkout@v5
+        with:
+          repository: ${{ inputs.main-repo }}
+          ref: ${{ inputs.main-ref }}
+          path: napari
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+          cache-dependency-path: |
+            napari/pyproject.toml
+      - name: Cache pooch downloads
+        uses: actions/cache@v4
+        with:
+          path: |
+            ${{ github.workspace }}/.cache/scikit-image
+            ${{ github.workspace }}/.cache/napari-*
+            ${{ github.workspace }}/.cache/pooch
+          key: pooch
+      - name: Setup headless display
+        uses: pyvista/setup-headless-display-action@v4.2
+        with:
+          qt: true
+          wm: herbstluftwm
```

Comment:
What does this line do?

Reply:
Never mind. It's a flag to install herbstulftwm window manager. True would have been more sensible. 🤷🏼